### ISSUE TITLE
Tic-tac-toe environment for the ES training process

### ISSUE THREAD

**Issue by Zeta36** [atgambardella/pytorch-es#4]
Hello, @atgambardella.

Using as base your code, I have developed a new Tic-tac-toe environment for the ES training process. As this game can be studied in full depth by a classical min-max tree, I've used this classic AI to play against our neural network model in the "step" phase and to return so the reward. 

The last result is a model (a simple "Linear" one) that thanks to the evolutionary computation can reach a zero-perfect game against the classical AI brute force strategy.

My code is here: https://github.com/Zeta36/pytorch-es-tic-tac-toe

I simplified also your code a little and I removed thing I knew I was not going to need.

Thanks for your work, friend.
